test/e2e/specs/treeview/00-dynamic-treeview.spec.js

```diff
@@ -0,0 +1,254 @@
+var deriva = require('../../utils/common/deriva-webapps');
+var httpFun = require('../../utils/common/httpFun.js');
+var testConfig = require('../../config');
+var expectedValue = require('./dynamic-treeview-expected-value.js');
+
+exports.tests = function (appName, baseUrl) {
+
+  describe('Dynamic page '+ appName, function () {
+
+    beforeAll(function () {
+      browser.ignoreSynchronization=true;
+      browser.get(baseUrl+ testConfig.treeviewPage.dynamic);
+      ele = element(by.id('rt-heading-Specimen&nbsp;Expression'));
+
+      deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
+    });
+
+    describe('legend panel', function () {
+      beforeAll(function () {
+        facet1 = deriva.treeviewPage.getListElements('facets-1');
+        facet2 = deriva.treeviewPage.getListElements('facets-2');
+        facet3 = deriva.treeviewPage.getListElements('facets-3');
+        facet4 = deriva.treeviewPage.getListElements('facets-4');
+        facet5 = element(by.css('#look-up > div > div:nth-child(5)'));
+
+      });
+
+      describe('strength section :', () => {
+
+        it('it should contain 6 options', function() {
+          facet1.count().then(function (count) {
+            expect(count).toEqual(6);
+          });
+        });
+
+        it('each list item should  have correct name', function() {
+          var facet1LinkNames = expectedValue.strength.names
+          facet1.all(by.css('span')).then(function (linkArr) {
+            linkArr.forEach(function (link) {
+              link.getText().then(function(text) {
+                 expect(facet1LinkNames).toContain(text);
+               });
+             });
+          });
+        });
+
+        it('each list item should  have correct image link', function(done) {
+          var facet1ImgLinks = expectedValue.strength.imgs;
+          imgFullLinks = facet1ImgLinks.map(function (ele) {
+            return baseUrl + ele;
+          });
+
+          facet1.all(by.tagName('img')).getAttribute('src').then(function (linkArr) {
+            linkArr.forEach(function (url) {
+               expect(imgFullLinks).toContain(url);
+               httpFun.httpGet(url).then(function (res) {
+                     expect(
+                       res.statusCode).toEqual(200, url + " is not a valid image source.");
+                     done();
+                   });
```

Comment:
In the places that you have added `done`, make sure to have a catch block that calls `done.fail()`. So in this case, it could look like:
```js
httpFun.httpGet(url).then(function (res) {
  expect(res.statusCode).toEqual(200, url + " is not a valid image source.");
  done();
}).catch(function (err) {
  console.log(err);
  done.fail();
});
```

This will make it easier to debug if there is an error in the `httpGet` function.